draft/implementation-notes/index.html

```diff
@@ -602,7 +602,154 @@ <h2>Forward Delta</h2>
         </li>
       </ul>
     </section>
+    
+    <section>
+      <h2>Guidance on Updating OCFL Object Versions</h2>
+      <p>
+      The OCFL is designed to be a specification that covers objects at rest and consequently
+      does not specify in detail update and file locking mechanisms since these are implementation
+      dependent features. Nevertheless, this section includes some basic guidance on how to 
+      update OCFL objects in a manner that tries to ensure that updates are limited to a single  
+      transaction and that failures are detectable and recoverable.
+      </p>
+    
+      <section>
+        <h2>Temporary Names</h2>
+        <p>  
+        While an OCFL object is being updated, and a new version is created, files and folders 
+        being written to should have temporary names that are not meaningful to OCFL parsers.
+        The top level inventory should also be updated last. This ensures that all the files 
+        referenced by the inventory are valid and, consequently, read-only clients that
+        reference the inventory should continue to operate normally except for the brief moment
+        when the inventory is actually being updated. In practice, it is expected that upstream
+        web caching should be able to cover this momentary unavailability in the majority of 
+        cases. The following temporary names are defined:
+	</p>	
+        <ul>
+          <li>  
+          Temporary Version Directories: New versions should be created in a temporary version directory.
+          This could reside in the OCFL Storage Root workspace directory ("deposit"), the same location as
+          other version directories in an OCFL Object or elsewhere. In any case, only one such directory 
+          can exist per OCFL Object at any time and updates to it should be managed by a single 
+          controlling process to avoid conflicts. It is advised that object updates should be given
+          a unique transaction identifier to simplify any overlying process control logic. For 
+          example, the identifier could be used as an an eTag for a REST API implementation. A 
+          good temporary directory name should be "deposit_transaction ID_OCFL Object ID". 
+          If there is guaranteed to only be a single process controlling updates to the contents 
+          of an OCFL storage root, then it is permissible to use "deposit_OCFL Object ID" as a 
```

Comment:
I wouldn't recommend directory names with spaces but maybe that is just old-school of me.

I think we also have to be careful around language like _permissible_ as this is non-normative, the issue here is that with just a single process it is _adequate_ without the transaction id.

Reply:
Yes, shouldn't be spaces - will fix that. How often will we ever have a single process controlling access since it is likely that any overlying API will be RESTful. 

This is in the implementation notes so not part of the spec and therefore shouldn't it all be non-normative. Adequate is adequate however!

Reply:
Mostly scrapped.